ToDoList/main.go

```diff
@@ -0,0 +1,185 @@
+// main.go
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+)
+
+func main() {
+	// 1) Define CLI flags, with some additional features -done, -undone and -clear
+	addPtr := flag.String("add", "", "Add a new to-do:       -add=\"Task description\"")
+	listPtr := flag.Bool("list", false, "List all to-do items:  -list")
+	updateIDPtr := flag.Int("update", 0, "ID of the to-do to update:  -update=<id> -desc=\"New description\"")
+	descPtr := flag.String("desc", "", "New description when using -update")
+	deleteIDPtr := flag.Int("delete", 0, "ID of the to-do to delete:  -delete=<id>")
+	clearAllPtr := flag.Bool("clear", false, "Delete ALL to-do items: -clear")
+	doneIDPtr := flag.Int("done", 0, "Mark a to-do as completed:  -done=<id>")
+	undoneIDPtr := flag.Int("undone", 0, "Unmark a completed to-do:  -undone=<id>")
+	flag.Parse()
+
+	// 2) Load existing todos from todos.json (internal storage evidence)
+	//First error handling practice
+	todos, err := loadTodos()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error loading todos: %v\n", err)
+		os.Exit(1)
+	}
+
+	// 3) If -clear is set, wipe the entire list and exit (added because it was annoying to have to delete each task one by one)
+	//Practicing a little error handling here
+	if *clearAllPtr {
```

Comment:
Since you are loading all your flags into pointers (essentially optional values), you can check them with a nil check rather than kind of relying on the default value which is what you do here. Same comment applies for all the other flags, just check for nil pointer rather than checking for zero/empty on the value.
```suggestion
	if clearAllPtr !=nil {
```